core/src/main/java/org/opensearch/sql/executor/execution/NonQueryPlan.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sql.executor.execution;
+
+import org.opensearch.sql.ast.tree.UnresolvedPlan;
+import org.opensearch.sql.common.response.ResponseListener;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.executor.QueryId;
+import org.opensearch.sql.executor.QueryService;
+
+/**
+ * Query plan which does not reflect a search query being executed.
+ */
+public class NonQueryPlan extends AbstractPlan {
```

Comment:
Is there a better way to call this than "non-query"?  
Maybe "CursorPlan"...?  Or "EmptyPlan"?  
I'm not really sure...

Reply:
yeah, I recall there are usually base class `Command` in other db for DDL or any statement that doesn't need planning. Otherwise many boilerplate code from AST to Physical will be added.

Reply:
"InertPlan"?